### ISSUE TITLE
http.so: undefined symbol: uidna_IDNToASCII

### ISSUE THREAD

**Issue by tweenlee** [oerdnj/deb.sury.org#1245]
**Describe the bug**
After upgraiding from pecl_http-3.2.0+2.6.0-1+ubuntu16.04.1+deb.sury.org+10 to pecl_http-3.2.1+2.6.0-1+ubuntu16.04.1+deb.sury.org+1
php7.1: symbol lookup error: /usr/lib/php/20160303/http.so: undefined symbol: uidna_IDNToASCII

**To Reproduce**
php7.1
 ```
<?php
$req = new \http\Client\Request("GET", "http://google.com", []);
$body = new \http\Message\Body();
$body->append("lala");
$req->setBody($body);
$client = new Client(null, null);
$client->enqueue($req);
while($client->once()) {
    $client->wait();
}
```
receive error: php7.1: symbol lookup error: /usr/lib/php/20160303/http.so: undefined symbol: uidna_IDNToASCII

**Distribution (please complete the following information):**
 - OS: Ubuntu Ubuntu 16.04.5 LTS
 - Architecture: 4.4.0-1092-aws #103-Ubuntu 9 x86_64 x86_64 x86_64 GNU/Linux
 - Repository: ppa:ondrej/php

**Package(s) (please complete the following information):**
php-pecl-http: 3.2.1+2.6.0-1+ubuntu16.04.1+deb.sury.org+1

**Additional context**
It works fine before the upgrade

**Comment by oerdnj** [comment]
What's the output of `ldd /usr/lib/php/20160303/http.so` on your system?

**Comment by tweenlee** [comment]
`ldd /usr/lib/php/20160303/http.so
    linux-vdso.so.1 =>  (0x00007ffdb7d0a000)
    libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f85c29a9000)
    libssl.so.1.1 => /usr/lib/x86_64-linux-gnu/libssl.so.1.1 (0x00007f85c271b000)
    libcurl.so.4 => /usr/lib/x86_64-linux-gnu/libcurl.so.4 (0x00007f85c24ac000)
    libevent-2.0.so.5 => /usr/lib/x86_64-linux-gnu/libevent-2.0.so.5 (0x00007f85c2266000)
    libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f85c2049000)
    libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f85c1c7f000)
    libcrypto.so.1.1 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007f85c17ae000)
    libidn.so.11 => /usr/lib/x86_64-linux-gnu/libidn.so.11 (0x00007f85c157b000)
    librtmp.so.1 => /usr/lib/x86_64-linux-gnu/librtmp.so.1 (0x00007f85c135f000)
    libssl.so.1.0.0 => /lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f85c10f6000)
    libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f85c0cb1000)
    libgssapi_krb5.so.2 => /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f85c0a67000)
    liblber-2.4.so.2 => /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2 (0x00007f85c0858000)
    libldap_r-2.4.so.2 => /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2 (0x00007f85c0607000)
    /lib64/ld-linux-x86-64.so.2 (0x00007f85c2e36000)
    libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f85c0403000)
    libgnutls.so.30 => /usr/lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007f85c00d3000)
    libhogweed.so.4 => /usr/lib/x86_64-linux-gnu/libhogweed.so.4 (0x00007f85bfea0000)
    libnettle.so.6 => /usr/lib/x86_64-linux-gnu/libnettle.so.6 (0x00007f85bfc6a000)
    libgmp.so.10 => /usr/lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f85bf9ea000)
    libkrb5.so.3 => /usr/lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007f85bf718000)
    libk5crypto.so.3 => /usr/lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007f85bf4e9000)
    libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007f85bf2e5000)
    libkrb5support.so.0 => /usr/lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f85bf0da000)
    libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f85beebf000)
    libsasl2.so.2 => /usr/lib/x86_64-linux-gnu/libsasl2.so.2 (0x00007f85beca4000)
    libgssapi.so.3 => /usr/lib/x86_64-linux-gnu/libgssapi.so.3 (0x00007f85bea63000)
    libp11-kit.so.0 => /usr/lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007f85be7ff000)
    libtasn1.so.6 => /usr/lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007f85be5ec000)
    libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007f85be3e8000)
    libheimntlm.so.0 => /usr/lib/x86_64-linux-gnu/libheimntlm.so.0 (0x00007f85be1df000)
    libkrb5.so.26 => /usr/lib/x86_64-linux-gnu/libkrb5.so.26 (0x00007f85bdf55000)
    libasn1.so.8 => /usr/lib/x86_64-linux-gnu/libasn1.so.8 (0x00007f85bdcb3000)
    libhcrypto.so.4 => /usr/lib/x86_64-linux-gnu/libhcrypto.so.4 (0x00007f85bda80000)
    libroken.so.18 => /usr/lib/x86_64-linux-gnu/libroken.so.18 (0x00007f85bd86a000)
    libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f85bd662000)
    libwind.so.0 => /usr/lib/x86_64-linux-gnu/libwind.so.0 (0x00007f85bd439000)
    libheimbase.so.1 => /usr/lib/x86_64-linux-gnu/libheimbase.so.1 (0x00007f85bd22a000)
    libhx509.so.5 => /usr/lib/x86_64-linux-gnu/libhx509.so.5 (0x00007f85bcfdf000)
    libsqlite3.so.0 => /usr/lib/x86_64-linux-gnu/libsqlite3.so.0 (0x00007f85bcd0a000)
    libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f85bcad2000)`

**Comment by tweenlee** [comment]
added reproducable code

**Comment by oerdnj** [comment]
For some reason, it didn't get linked to libicu, it's time to switch to libidn2 anyway.

**Comment by drkskwlkr** [comment]
> it's time to switch to libidn2 anyway.

Caution with switching to libidn2 atm, see https://github.com/oerdnj/deb.sury.org/issues/1247

**Comment by oerdnj** [comment]
I believe the issue has been resolved in 3.2.1+2.6.0-2
